biogeochem/FatesSoilBGCFluxMod.F90

```diff
@@ -148,18 +145,19 @@ function GetPlantDemand(ccohort,element_id) result(plant_demand)
        return
     end if
        
-
+    fnrt_c = ccohort%prt%GetState(fnrt_organ, carbon12_element)
+    
     ! If the plant is not a newly recruited plant
     ! We use other methods of specifying nutrient demand
     ! -----------------------------------------------------------------------------------
 
     if(element_id.eq.nitrogen_element) then
 
-       plant_demand = smth_fac*ccohort%daily_n_demand + (1._r8-smth_fac)*max(0._r8,ccohort%daily_n_need)
+       plant_demand = fnrt_c * EDPftvarcon_inst%vmax_nh4(ccohort%pft) * sec_per_day
```

Comment:
Good question. RD wants the plants to present a joint unified demand on nitrogen uptake. It does this because the demand is used to drive NH4 uptake, and then the remainder is applied to NO3.  Thus with only one demand, we use only one parameter vmax_nh4.  

However, we don't have to do it this way.  ECA needs the vmax differentiated for NH4 and NO3.  It might be cleaner if we have the vmax parameters the same for both RD and ECA. But, in the case of RD, we will simply add them together...?

Reply:
Yeah, I was expecting to see the sum of NO3 and NH4 Vmax multiplied by the root mass to get the whole plant demand. Just based on the concept. 

That would increase plant competitiveness versus microbes in RD. I'm not sure what that balance looks like currently. But with RD in the FACE simulations, plants take up equivalent levels of N to ECA at much lower Vmax values, at least an order of magnitude lower Vmax if not more. That makes sense as RD and ECA are fundamentally different approaches, it's unlikely that with ECA N uptake is operating near Vmax for much of the time. 

So while conceptually it makes sense to add them, it will further increase plant competitiveness and further widen the gap between RD and ECA uptake magnitudes for the same Vmax values.

Reply:
Yeah, the meaning of the vmax parameters for RD are much different than with ECA.  Right now there is just no way around it, because RD applied the same demand sequentially to the NH4 and then to the NO3.  Whereas ECA applies the two different demands NH4 and NO3 separately and exclusively to their parameters. 

After this PR, we can re-evaluate how we form the NO3 demand from FATES to RD.  We could apply the demands in RD independantly.  But I would like to do it as a separate thing because it would require some non-trivial coding, particularly because that code assumes there is only 1 joint n demand term.